case_studies/dc2_cataloging/utils/encoder.py

```diff
@@ -109,12 +84,16 @@ def _compute_loss(self, batch, logging_name):
             band=self.reference_band,
         )
 
-        x = self.image_normalizer.get_input_tensor(batch)
-        x_features = self.features_net(x)
-
+        x_features = self.get_features(batch)
         x_cat = self.catalog_net(x_features)
         loss = self.var_dist.compute_nll(x_cat, target_cat)
 
+        nan_mask = torch.isnan(loss)
+        if nan_mask.any():
+            loss = loss[~nan_mask]
+            msg = f"NaN detected in loss. Ignored {nan_mask.sum().item()} NaN values."
+            warnings.warn(msg)
```

Comment:
Has this helped? I deleted it from the main encoder because although it seemed like a good idea, it never seemed to catch any nans.

Reply:
In my experiments, it also can't catch any NaN.